File: examples/c/write.c

#include <fcntl.h> /* open, O_WRONLY */
#include <unistd.h> /* write, close */
#include <stdio.h> /* perror */
#include <stdlib.h> /* EXIT_SUCCESS, EXIT_FAILURE */

int main(void)
{
	ssize_t bytes_written = 0;
	int port_fd = 0;
	char data[] = "Hello world!";

	port_fd = open("/dev/fscc0", O_WRONLY);

	if (port_fd == -1) {
		perror("open");
		return EXIT_FAILURE;
	}

	bytes_written = write(port_fd, data, sizeof(data));
	
	close(port_fd);
	
	return EXIT_SUCCESS;
}

File: examples/c#/write.cs
using System;
using System.IO;

namespace FSCC
{
	class MainClass
	{
		public static void Main(string[] args)
		{
		    FSCC.Port port = null;
		    
		    try
		    {
			    port = new Port("/dev/fscc0", FileAccess.Write);
			}
			catch (System.UnauthorizedAccessException e)
			{
			    Console.WriteLine(e.Message);
			    
			    return;
			}
					
			port.Write("Hello world!");
			
			port.Close();
		}
	}
}
